app/scripts/controllers/transactions/pending-tx-tracker.js

```diff
@@ -157,8 +157,19 @@ export default class PendingTransactionTracker extends EventEmitter {
 
       return
     }
-    // *note to self* hard failure point
-    const transactionReceipt = await this.query.getTransactionReceipt(txHash)
+
+    let transactionReceipt
+    try {
+      transactionReceipt = await this.query.getTransactionReceipt(txHash)
+    } catch {
+      const err = new Error('Missing transaction receipt')
+      txMeta.warning = {
+        error: err.message,
+        message: 'There was a problem loading this transaction.',
+      }
+      this.emit('tx:warning', txMeta, err)
+      return
```

Comment:
In the case where the nonce was taken (e.g. where the `taken` variable below was set to `true`), this will now return early and emit a `tx:warning` instead of emitting `tx:dropped`.

This could be addressed by checking to see whether the nonce was taken before getting the transaction receipt. It would also mean the `try` block below wouldn't be needed anymore.